src/app/lib/profiles/profile.mainsail-migrator.ts

```diff
@@ -31,6 +39,16 @@ export class ProfileMainsailMigrator implements IProfileMainsailMigrator {
 
 			// If this public key has already been migrated, skip to avoid duplicates
 			if (publicKey !== undefined && seenPublicKeys.has(publicKey)) {
+				const mergedWallet = Object.values(wallets).find(
+					(d) => d.data["PUBLIC_KEY"] === publicKey && migratedWallets[d.id] !== undefined,
```

Comment:
A minor thing. You an use WalletData.PublicKey enum for the `PUBLIC_KEY`
```suggestion
					(d) => d.data[WalletData.PublicKey] === publicKey && migratedWallets[d.id] !== undefined,
```

Reply:
Fixed it, thanks for the suggestion.